graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/fieldtypes/streams/StreamsForFieldRetrieverES7.java

```diff
@@ -47,42 +48,43 @@ public StreamsForFieldRetrieverES7(final ElasticsearchClient client) {
         this.client = client;
     }
 
+    private record FieldBucket<T>(String fieldName, T value) {
```

Comment:
Sorry if I am missing smth, but is it really needed to introduce a second, generic type called R ?
Cannot we simplify it to:
```
    private record FieldBucket<T>(String fieldName, T value) {
        FieldBucket<T> withValue(T newValue) {
            return new FieldBucket<>(fieldName, newValue);
        }
    }
```

TBH, I'd even consider removing `withValue` method completely, the constructor is so simply that it probably does not make that much sense to wrap it...

Reply:
> Sorry if I am missing smth, but is it really needed to introduce a second, generic type called R ?
Yes, it will return an instance with a different type for `value`. If you use `T`, then you can only pass a value of the same type the instance was originally instantiated with.

> TBH, I'd even consider removing withValue method completely, the constructor is so simple that it probably does not make that much sense to wrap it...

I think it provides value, because a) it ensures that the same `fieldName` is kept and b) because the `withProperty`-pattern is very common for value classes and transports the intention quite well.